paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/ResourceFolderRepository.kt

```diff
@@ -75,4 +86,160 @@ class ResourceFolderRepository(
     if (namespace != this.namespace) null else resourceTable[resourceType]
 
   override fun getNamespace(): ResourceNamespace = namespace
+
+  /**
+   * Inserts the given resources into this repository.
+   */
+  private fun commitToRepository(itemsByType: Map<ResourceType, ListMultimap<String, ResourceItem>>) {
+    for ((key, value) in itemsByType) {
+      getOrCreateMap(key).putAll(value)
+    }
+  }
+
+  private fun getOrCreateMap(type: ResourceType): ListMultimap<String, ResourceItem> =
+    // Use LinkedListMultimap to preserve ordering for editors that show original order.
+    resourceTable.computeIfAbsent(type) { LinkedListMultimap.create() }
+
+  private fun checkResourceFilename(file: PathString): Boolean {
+    return true
+    // return myPsiNameHelper.isIdentifier(SdkUtils.fileNameToResourceName(file.fileName))
+  }
```

Comment:
@jrodbx I am having trouble to port PsiNameHelper over, let us discuss in tomorrow sync meeting